*** drm/configure.ac	2009-03-29 14:41:48.000000000 +0400
--- drm/configure.ac	2009-03-29 14:28:50.000000000 +0400
***************
*** 62,67 ****
--- 62,79 ----
    AC_MSG_RESULT([$libdrm_cc_flag])
  ])
  
+ dnl We use the system's libdrm for the test programs
+ 
+ AC_CHECK_FUNCS([drmGetVersion], [DRM_LIB=],
+                           [AC_CHECK_LIB([drm], [drmGetVersion], [DRM_LIB=-ldrm], 
+                                                         [AC_MSG_ERROR([Couldn't find libdrm])])])
+ AC_SUBST([DRM_LIB])
+ 
+ AC_CHECK_FUNCS([drm_mmInit], [DRM_INTEL_LIB=],
+                           [AC_CHECK_LIB([drm_intel], [drm_mmInit], [DRM_INTEL_LIB=-ldrm_intel], 
+                                                         [AC_MSG_ERROR([Couldn't find libdrm_intel])])])
+ AC_SUBST([DRM_INTEL_LIB])
+ 
  dnl Use lots of warning flags with with gcc and compatible compilers
  
  dnl Note: if you change the following variable, the cache is automatically
