*** drm/tests/modetest/Makefile.am~	2009-03-30 12:47:56.000000000 +0400
--- drm/tests/modetest/Makefile.am	2009-03-30 12:49:44.000000000 +0400
***************
*** 10,15 ****
  modetest_SOURCES = \
  	modetest.c
  modetest_LDADD = \
! 	$(top_builddir)/libdrm/libdrm.la \
! 	$(top_builddir)/libdrm/intel/libdrm_intel.la \
  	$(CAIRO_LIBS)
--- 10,14 ----
  modetest_SOURCES = \
  	modetest.c
  modetest_LDADD = \
! 	@DRM_LIB@ @DRM_INTEL_LIB@
  	$(CAIRO_LIBS)
