*** drm/tests/Makefile.am	2009-03-20 03:24:48.000000000 +0300
--- drm/tests/Makefile.am	2009-03-29 14:46:00.000000000 +0400
***************
*** 10,17 ****
  libdrmtest_la_SOURCES = \
  	drmtest.c \
  	drmtest.h
! libdrmtest_la_LIBADD = \
! 	$(top_builddir)/libdrm/libdrm.la
  
  LDADD = libdrmtest.la
  
--- 10,16 ----
  libdrmtest_la_SOURCES = \
  	drmtest.c \
  	drmtest.h
! libdrmtest_la_LIBADD = @DRM_LIB@ @CLOCK_LIB@
  
  LDADD = libdrmtest.la
  
