*** drm/tests/modetest/modetest.c	2009-03-20 03:24:48.000000000 +0300
--- drm/tests/modetest/modetest.c	2009-03-29 15:28:56.000000000 +0400
***************
*** 45,53 ****
  #include <string.h>
  #include <errno.h>
  
! #include "xf86drm.h"
! #include "xf86drmMode.h"
! #include "intel_bufmgr.h"
  
  drmModeRes *resources;
  int fd, modes;
--- 45,53 ----
  #include <string.h>
  #include <errno.h>
  
! #include <xf86drm.h>
! #include <xf86drmMode.h>
! #include <intel_bufmgr.h>
  
  drmModeRes *resources;
  int fd, modes;
